### ISSUE TITLE
The ASM band-limit condition doesn't consider the case when kykx is not zero

### ISSUE THREAD

**Issue by Wenyi-lee** [chromatix-team/chromatix#180]
The issue appears in /src/chromatix/functional/propagation.py line 575.

Currently:        H_filter_yx = jnp.abs(field.f_grid - k0) <= k_max

If kykx !=0, then field.f_grid should be modified by kykx to represent the correct current field frequency grid.

**Comment by Wenyi-lee** [comment]
I double check the code with the paper shifted ASM one, seems code is correct. But the calculation result is a bit wired to me when the field propagates long with a large angle. Maybe the reason is the bandlimit is too strong for the case, maybe this paper would help to make the bandwidth to be extended [https://doi.org/10.1364/OE.419096](https://doi.org/10.1364/OE.419096)

**Comment by diptodip** [comment]
Sorry it took me a while to look at this and thanks for the pointer! I'll see if BEASM works out. Do you have an example handy where the result looks weird to you?

**Comment by Wenyi-lee** [comment]
I am sorry for the late reply. 

I reread the paper and i think your implementation is correct according all the equations in the paper (Shifted angular spectrum method for off-axis numerical propagation).

The weird result to me was when i turn on the bandlimit to propagate a field 400 um* 400 um with 10 degree angle and propagate about 2-3 mm, the bandlimited window filter the frequency a bit weird than the real world. Looks like it doesn't filter at the center of the frequency carried by the tilted wave as shown below.

<img width="1372" height="748" alt="Image" src="https://github.com/user-attachments/assets/3c48c408-d14d-4ac0-838f-85e46094e9b5" />

 I didn't understand this situation for a while, and then i found this paper. ([Modified shifted angular spectrum method for numerical propagation at reduced spatial sampling rates](https://doi.org/10.1364/OE.22.026265) )

In their comparison, they analysed that the sampling condition with the shifted-ASM is still the same as the origianl ASM, so if you are undersampling the field for the tilt propagation than the requirement of ASM for the propagation angle, then you will get an alising field. I feel this might be the case for me. All the code is correct for shiffted-ASM. But what i need is this modified-shifted-ASM.

I will do some tests, and will let you know if i find anything else.

**Comment by diptodip** [comment]
No worries. And great, I think this modified shifted ASM also looks easy enough to add if this handles your example. What was the sampling of this field (how many pixels)?

**Comment by Wenyi-lee** [comment]
The sampling pitch is 1um, for 1um wavelength, with padding of 2 times of the field (also tried 4 times). I guess this is not enough for using shifted-ASM correctly for my case. But i cannot afford sampling it too much as it is part of the optimization.
